### PR TITLE
CMR-7078 index direct distribution information in elasticsearch

### PR DESCRIPTION
**Pull request by jaybarra**

### CODE REVIEW

File: system-int-test/test/cmr/system_int_test/ingest/collection_direct_distribution_fields_test.clj [hunk 1]
```diff
@@ -0,0 +1,65 @@
+(ns cmr.system-int-test.ingest.collection-direct-distribution-fields-test
```
Comment (ygliuvt) [inline system-int-test/test/cmr/system_int_test/ingest/collection_direct_distribution_fields_test.clj:1]: No need to add this test. We will do the testing when the search ticket is worked.

File: indexer-app/src/cmr/indexer/data/index_set.clj [hunk 2]
```diff
@@ -218,6 +218,13 @@
    :uuid m/string-field-mapping
    :uuid-lowercase m/string-field-mapping})
 
+(defnestedmapping direct-distribution-information-mapping
```
Comment (ygliuvt) [inline indexer-app/src/cmr/indexer/data/index_set.clj:221]: I don't see the need to index this as a nested document.

**Review by ygliuvt — CHANGES_REQUESTED** [review]
Don't see the need for nested index mapping

File: system-int-test/src/cmr/system_int_test/utils/index_util.clj [hunk 3]
```diff
@@ -168,6 +168,26 @@
...
   [coll]
   (client/post (format "%s/1_small_collections/_delete_by_query" (url/elastic-root))
-                 {:connection-manager (s/conn-mgr)
-                  :body (query-for-granules-by-collection coll)
-                  :content-type "application/json"}))
+               {:connection-manager (s/conn-mgr)
+                :body (query-for-granules-by-collection coll)
+                :content-type "application/json"}))
+
+(defn get-collection-es-doc-by-concept-id
```
Comment (ygliuvt) [inline system-int-test/src/cmr/system_int_test/utils/index_util.clj:175]: Don't see this function getting used anywhere. Delete it.

File: indexer-app/src/cmr/indexer/data/concepts/collection.clj [hunk 4]
```diff
@@ -187,6 +199,8 @@
          publication-references :PublicationReferences
          collection-citations :CollectionCitations} collection
         parsed-version-id (collection-util/parse-version-id version-id)
+        direct-distribution-information (umm-c-ddi->es-doc-ddi
```
Comment (ygliuvt) [inline indexer-app/src/cmr/indexer/data/concepts/collection.clj:202]: No need to retrieve the unused values from UMM-C.

Comment (jaybarra) [reply indexer-app/src/cmr/indexer/data/concepts/collection.clj:202]: They need to be converted someplace. The s3-bucket-and-prefix-names value doesn't come in by itself, it's part of a DirectDistrubtionInformationType.

Comment (ygliuvt) [reply indexer-app/src/cmr/indexer/data/concepts/collection.clj:202]: We don't need all the values in DirectDistrubtionInformationType, you can remove `umm-c-ddi->es-doc-ddi` function and just parse out `s3-bucket-and-object-prefix-names` here.

**Review by sxu123 — APPROVED** [review]

**Review by ygliuvt — APPROVED** [review]

**Pull request merged**